internal/helm/getter/client_opts_test.go

```diff
@@ -196,6 +196,118 @@ func Test_tlsClientConfigFromSecret(t *testing.T) {
 	}
 }
 
+func Test_registryTLSoginOption(t *testing.T) {
+	tlsCA, err := os.ReadFile("../../controller/testdata/certs/ca.pem")
+	if err != nil {
+		t.Errorf("could not read CA file: %s", err)
+	}
+
+	tests := []struct {
+		name       string
+		certSecret *corev1.Secret
+		authSecret *corev1.Secret
+		wantNil    bool
```

Comment:
could be better to have something like `noOfOpts int` and then exactly compare it to `clientOpts.RegLoginOpts`